docs/reference.rst

```diff
@@ -0,0 +1,66 @@
+*********
+Reference
+*********
+
+Rockcraft.yaml format
+=====================
+
+A Rockcraft project is defined in a YAML file named ``rockcraft.yaml``
+at the root of the project tree in the filesystem.
+
+
+Format specification
```

Comment:
We can probably leverage the pydantic model here for some autodocumentation of the mode.